src/events.ts

```diff
@@ -114,14 +112,24 @@ export interface SeekEvent extends Event {
    */
   from: {
     time: number;
-    source: Source;
+    source: {
+      duration: number;
+      isActive: boolean;
+      isAttachedToPlayer: boolean;
+      metadata?: Record<string, any>;
+    };
```

Comment:
About the changes related to the `Source` type, before this PR, it was simply an [interface](https://www.typescriptlang.org/docs/handbook/2/objects.html) that described the shape of the object that's being passed from native to JS. But now, `Source` is a class type wrapping the internals of a native `Source` object and not simply a type hint like before (properties are accessed through async functions for example), so I couldn't use it here for this purpose.

If using the literal object type notation here seems a problem to you, I can change it to a type alias. What do you think?

Reply:
Yes, I think that would make sense. Imagine we change something (rename a property, add another property, etc.), then we would need to touch all the events again.
I am not very familiar with TS unfortunately. I would suspect that the `Source` implementation could just inherit from the interface? If that's not possible a type alias is probably ok.

Reply:
Unfortunately, it cannot inherit from this interface because the `Source` implementation uses mainly async methods to fetch properties of a native `Source`. After all, React's bridge is async. See [`Source#duration`](https://github.com/bitmovin/bitmovin-player-react-native/blob/feature/drm-playback/src/source.ts#L130) for example.

By the way, here's the commit that adds the interface: 709e04cc225c7f54dfbefe9cad41e30927b3e076

Reply:
I see. For the mobile SDKs we provide a reference to the object. That way it is possible to directly use the `Source` to "do additional work".
The way it is in the bridge now, is that the `Source` in the events is only a snapshot of the actual `Source` at the time the event was fired. 

If that's the only way to model this, then this is of course fine. Having von definition is already a great improvement over defining it in every event again.

Reply:
Yeah, unfortunately, that's the only way right now to access the source object that comes within events. But for every other use of source, it's possible to access its native reference and do additional work just like the mobile SDKs.